pkg/controllers/node/termination/controller.go

```diff
@@ -138,6 +152,56 @@ func (c *Controller) removeFinalizer(ctx context.Context, n *v1.Node) error {
 	return nil
 }
 
+func (c *Controller) terminationGracePeriodExpirationTime(ctx context.Context, node *v1.Node) (*time.Time, error) {
```

Comment:
A thought: What if rather than introspecting our owner, we drove this through an annotation? The NodeClaim termination controller adds an annotation that gives this controller a `karpenter.sh/force-terminate-by` annotation that informs us when we have to completely terminate the node by.

Reply:
That was the design for the initial implementation I worked on before I was recommended to use `deletionTimestamp` instead for simplicity. :smile:

Reply:
Thinking about this more, I could avoid the extra NodeClaim lookup by checking for `deletionTimestamp` in the nodeclaim controller, right?